plugins/module_utils/rds.py

```diff
@@ -156,21 +180,10 @@ def call_method(client, module, method_name, parameters):
     changed = True
     if not module.check_mode:
         wait = module.params.get('wait')
-        # TODO: stabilize by adding get_rds_method_attribute(method_name).extra_retry_codes
+        retry_codes = get_rds_method_attribute(method_name, module).retry_codes
         method = getattr(client, method_name)
         try:
-            if method_name in ['modify_db_instance', 'promote_read_replica']:
-                # check if instance is in an available state first, if possible
-                if wait:
-                    wait_for_status(client, module, module.params['db_instance_identifier'], 'modify_db_instance')
-                result = AWSRetry.jittered_backoff(catch_extra_error_codes=['InvalidDBInstanceState'])(method)(**parameters)
-            elif method_name == 'modify_db_cluster':
-                # check if cluster is in an available state first, if possible
-                if wait:
-                    wait_for_status(client, module, module.params['db_cluster_identifier'], method_name)
-                result = AWSRetry.jittered_backoff(catch_extra_error_codes=['InvalidDBClusterStateFault'])(method)(**parameters)
-            else:
-                result = AWSRetry.jittered_backoff()(method)(**parameters)
```

Comment:
I'm not sure if we want to delete all these. If we delete them, it seems we don't catch anymore the catch_extra_error_codes because we don't handle those methods inside call_method.

Reply:
@alinabuzachis we still do! its just using the get_rds_method_attribute to get the error codes instead of in this method - there are integration tests that check for the error messages

Reply:
get_rds_method_attribute does not handle  the methods you deleted here, don't understand how you get those retries when the method will be called by rds_instance and rds_cluster.